### ISSUE TITLE
Compilations

### ISSUE THREAD

**Issue by hinto-janai** [hinto-janai/festival#49]
## Feature
Compilation support.

A single `Album` with varying `Artist`'s.

Metadata tag will most likely be `Compilation` and the proprietary `itunescompilationflag`.

## Problem
There _must_ be an owner of this `Album`, aka an `Artist`.

Within this compilation `Album` struct, what `Artist` will the `album.artist` point to?

It's unclear in a compilation who "owns" the `Album`.

**Comment by jansol** [comment]
As far as I've seen usually, there is a separate AlbumArtist tag that is used for the album itself, regardless of whether it is flagged as a compilation. Each individual track in the album can then have its own Artist tag independently of that. The "owner" of the album would be the AlbumArtist.

**Comment by hinto-janai** [comment]
Yes, that's actually the first tag Festival looks for when looking for an artist.

There's not really a standard though, some compilations won't have the same `AlbumArtist` in every song, some won't have an `Artist` at all (`Composer`, `Performer`, etc).

**Comment by jansol** [comment]
I'd argue that compilations with inconsistent `AlbumArtist` are simply mistagged or the files are pulled from different albums and just slapped in a folder together for convenience. In general I'd model the metadata information fairly closely to [MusicBrainz model & style](https://musicbrainz.org/doc/Style) since that is documented in great detail, which makes it much easier to achieve predictable behaviour across different programs. I also haven't seen any other (documented) scheme with comparable accuracy & flexibility.

Opinions will of course vary :shrug: The full model is quite involved after all, and only a fraction of it actually maps to well-known tag names in files. At least `*-mbid` tags are somewhat standard, so it is easy to pull all additional information from the web when those tags are present.

I guess a sensible solution would be to lump tracks together by folder, additionally creating separate albums for each `AlbumArtist`, or if that is missing, for each `Artist` in the folder. Maybe even try `Performer` and `Composer` as fallbacks if you deem it necessary. And if there's no artist information, just fall back to "Unknown Artist" for the entire folder. But tbh I'd question whether that isn't already way more effort than it is worth. In the end it's just a heuristic, and the correct solution would be to just fix the tags in the first place. For people who just want to have arbitrary collections I'd consider letting them specify a file path template that specifies the layout of their library, something like `${Artist}/${Year}-${Album}/${trackno}-${Title}`
